tools/executable_analysis_tools/elf-float-checker.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Script for checking for floating point symbols in ELF files."""
+
+import argparse
+import logging
+import re
+import subprocess
+import sys
+
+log = logging.getLogger(__name__)
+
+LOG_FORMAT = "%(asctime)s - %(levelname)s - %(message)s"
+FLOATING_POINT_SYMBOL_REGEX = r"__aeabi_(cd.+|cf.+|h2f.+|d.+|f.+|.+2d|.+2f)"
+
+
+class SymbolParser:
+    """Parse the given ELF format file."""
+
+    def get_symbols_from_table(self, symbol_table, symbol_regex):
+        """Get symbols matching a regular expression pattern from a table."""
+        pattern = re.compile(symbol_regex, re.X)
+        matched_symbols = []
+        symbol_table_lines = symbol_table.split("\n")
+        for symbol_table_line in symbol_table_lines:
+            match = pattern.search(symbol_table_line)
+            if match:
+                log.debug("Symbol line: {}".format(symbol_table_line))
+                log.debug("Match found: {}".format(match))
+                matched_symbols.append(match.group(0))
+
+        log.debug("Symbols found:\n'{}'".format(matched_symbols))
+        return matched_symbols
+
+    def get_symbol_table(self, elf_file):
+        """Get the symbol table from an ELF format file."""
+        log.debug(
+            "Get the symbol table for ELF format file '{}'".format(elf_file)
+        )
+        # Command syntax:
+        # `arm-none-eabi-readelf --symbols <elf_file>`
+        cmd = ["arm-none-eabi-readelf", "--symbols", elf_file]
```

Comment:
use `objdump -t` instead for portability

Reply:
Can you give a bit more information what you think `objdump` is more portable than `readelf`?

Reply:
CI doesn't have the latter

Reply:
d317feb